### PR TITLE
[spec] Improve value sequence docs

### PR DESCRIPTION
**Pull request by ntrel**
Prefer using TypeSeq for variadic template functions. Mention returning `Tuple`.
Add subheading and example for lvalue sequences.
Minor tweaks.

### CODE REVIEW

File: spec/template.dd [hunk 1]
```diff
@@ -933,10 +933,46 @@ $(H4 $(LNAME2 homogeneous_sequences, Homogeneous Sequences))
         ---
         )
 
+    $(BEST_PRACTICE Prefer using a *TypeSeq* to declare variadic template
```
Comment (dkorpel) [inline spec/template.dd:936]: Prefer a TypeSeq over what exactly?

File: spec/template.dd [hunk 2]
```diff
@@ -933,10 +933,46 @@ $(H4 $(LNAME2 homogeneous_sequences, Homogeneous Sequences))
...
+        functions because then there is only one template instantiation for
+        each set of types.)
```
Comment (dkorpel) [inline spec/template.dd:938]:
```suggestion
        each list of types.)
```

A different order still gives a different template instantiation

File: spec/template.dd (hunk 1 above)
Comment (PetarKirov) [reply spec/template.dd:936]: I also don't understand what the recommendation means.

Comment (ntrel) [reply spec/template.dd]: I've now removed this paragraph.

**Review by dkorpel — APPROVED** [review]

**Pull request merged**